### PR TITLE
Use chromium_src for vector_icons overrides.

### PR DESCRIPTION
**Pull request by goodov**
Reimplement how `vector_icons` overrides are applied. The updated logic is simpler, has less patch points and uses the general chromium_src approach.

Resolves https://github.com/brave/brave-browser/issues/40355

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by goodov** [comment]
@brave/chromium-src-reviewers @brave/build-commands-reviewers ptal.

File: chromium_src/components/vector_icons/aggregate_vector_icons.py [hunk 1]
```diff
@@ -4,60 +4,90 @@
...
+        chromium_src_override = brave_chromium_utils.get_chromium_src_override(
+            icon_path)
+
+        # Leo override may have a different name than the original icon.
+        # Copy the override to gen/brave/vector_icons_overrides with the
+        # original name.
+        leo_override = leo_overrides.get(icon_ws_path)
+        if leo_override:
+            if os.path.exists(chromium_src_override):
+                raise RuntimeError(
```
Comment (fallaciousreasoning) [inline chromium_src/components/vector_icons/aggregate_vector_icons.py:59]: :+1:

File: chromium_src/components/vector_icons/aggregate_vector_icons.py [hunk 2]
```diff
@@ -4,60 +4,90 @@
...
-def get_root():
-    return f'{Path(__file__).parent}/../..'
+VERBOSE = False  # Set to True to print verbose messages.
 
 
-def get_root_relative(icon_path):
-    return "//" + os.path.relpath(os.path.abspath(icon_path), get_root()) \
-      .replace(os.sep, '/')
+# Loads the leo overrides from leo_overrides.json.
+def LoadLeoOverrides():
```
Comment (fallaciousreasoning) [inline chromium_src/components/vector_icons/aggregate_vector_icons.py:18]: Hey, not sure if we care or not, but I don't think Chromium uses CamelCase any more in their python style:
https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/python/python.md#our-previous-python-style

They say to keep using it in existing files which use it, but going forward to use `unix_hacker_style`

**Review by fallaciousreasoning — APPROVED** [review]
lgtm with a nit (which might not be a problem)

File: chromium_src/components/vector_icons/aggregate_vector_icons.py [hunk 3]
```diff
@@ -4,60 +4,90 @@
...
+        # Copy the override to gen/brave/vector_icons_overrides with the
+        # original name.
+        leo_override = leo_overrides.get(icon_ws_path)
+        if leo_override:
+            if os.path.exists(chromium_src_override):
+                raise RuntimeError(
+                    f'leo override and chromium_src override both exist for '
+                    f'{icon_path}')
+
+            gen_file = f'gen/brave/vector_icons_overrides/{icon_ws_path[2:]}'
```
Comment (atuchin-m) [inline chromium_src/components/vector_icons/aggregate_vector_icons.py:63]: nit: make `gen/brave/vector_icons_overrides/` a constant.

**Review by atuchin-m — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

**Pull request merged**